app/models/financialDetails/FinancialDetailsResponseModel.scala

```diff
@@ -91,7 +93,7 @@ case class FinancialDetailsModel(documentDetails: List[DocumentDetail],
   }
 
   def merge(financialDetailsModel: FinancialDetailsModel): FinancialDetailsModel = {
-    FinancialDetailsModel(documentDetails ++ financialDetailsModel.documentDetails,
+    FinancialDetailsModel(balanceDetails, documentDetails ++ financialDetailsModel.documentDetails,
```

Comment:
Debatable: this function does not merge `balanceDetails` anyhow. Would it be more feasible to rename it to something like `def mergeLists`?

Reply:
Agreed and made those changes.